### PR TITLE
export xp_call_name() for downstream use

### LINKED ISSUE #2023: Export xp_call_name()?
Here is another very helpful companion to `xml_nodes_to_lints()` that would be nice for authors writing custom linters. I find myself using `lintr:::xp_call_name()` now.

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #2023

### CODE REVIEW

**Comment by AshesITR** [comment]
Wondering about if the name is "nice" for exporting.

* Is `xp_` a common prefix for any package?
* Is there a package for XPaths that has set a standard for XPath-related functions like xml2 did for `xml_`?

**Comment by MichaelChirico** [comment]
> Is `xp_` a common prefix for any package?

No. Here's the 3 packages (including {lintr}) on CRAN with any functions starting with `xp_`:

https://github.com/search?q=org%3Acran+language%3Ar+%2F%5Exp_%5Ba-zA-Z0-9._%5D%2B%5Cs*%28%3C-%7C%3D%29%5Cs*function%2F&type=code

 - {dndR} is for Dungeons & Dragons, XP = Experience Points
 - {xpose} I think uses it for "xplot" (not sure what that is)

For packages using `_?xp_` anywhere in their NAMESPACE, see:

https://github.com/search?q=org%3Acran+path%3ANAMESPACE+%2F%5B%5Ea-zA-Z%5Dxp_%5Ba-zA-Z0-9._%5D%2B%2F&type=code

> Is there a package for XPaths that has set a standard for XPath-related functions like xml2 did for xml_?

AFAIK, no. The only other CRAN package mentioning "XPath" in its title is {selectr}, which is for translating CSS queries to XPath (and not the other way around).

https://cran.r-project.org/web/packages/available_packages_by_name.html

There are many more packages that use "XPath" somewhere in their sources, I didn't see any pattern emerge when glancing through the results here:

https://github.com/search?q=org%3Acran+language%3Ar+%22xpath%22&type=code

Here are all CRAN functions using "xpath"... still not solid pattern:

https://github.com/search?q=org%3Acran+language%3Ar+%2F%28%5E%7C%5B%5Ea-zA-Z0-9%5D%29xpath%5Ba-zA-Z0-9._+%3C%3D-%5D%2Bfunction%2F&type=code

**Comment by AshesITR** [comment]
Thanks for doing the research. WDYT about `xml_call_name()`?

**Comment by MichaelChirico** [comment]
> WDYT about xml_call_name()?

Not strongly opposed. Main worry is confusion vs {xml2} functions. Esp. if this grows to be a family of exported functions (there are already a few more private `xp_*` functions).

We also use `xmlparsedata::xml_parse_data()`. There, `xml_parse` could be the prefix (since they also export `xml_parse_token_map`) & matches the package name, so it's not as collided.

File: NEWS.md [hunk 1]
```diff
@@ -30,6 +30,8 @@
 * `conjunct_test_linter()` also lints usage like `dplyr::filter(x, A & B)` in favor of using `dplyr::filter(x, A, B)` (part of #884, @MichaelChirico).
 * `sort_linter()` checks for code like `x == sort(x)` which is better served by using the function `is.unsorted()` (part of #884, @MichaelChirico).
 * `paste_linter()` gains detection for file paths that are better constructed with `file.path()`, e.g. `paste0(dir, "/", file)` would be better as `file.path(dir, file)` (part of #884, @MichaelChirico).
+* `fixed_regex_linter()` gains an option `allow_unescaped` (default `FALSE`) to toggle linting regexes not requiring any escapes or character classes (#1689, @MichaelChirico). Thus `fixed_regex_linter(allow_unescaped = TRUE)` would lint on `grepl("[$]", x)` but not on `grepl("a", x)` since the latter does not use any regex special characters.
```
Comment (AshesITR) [inline NEWS.md:33]: This diff looks sporadic

File: R/xp_utils.R [hunk 2]
```diff
@@ -44,7 +44,41 @@ xp_and <- function(...) paren_wrap(..., sep = "and")
 #' @noRd
 xp_or <- function(...) paren_wrap(..., sep = "or")
 
+#' Get the name of the function matched by an XPath
+#'
+#' Often, it is mroe helpful to custom-tailer the `message` of a lint to record
```
Comment (AshesITR) [inline R/xp_utils.R:49]: typo

File: R/xp_utils.R [hunk 3]
```diff
@@ -44,7 +44,41 @@ xp_and <- function(...) paren_wrap(..., sep = "and")
...
+#'   which function was matched by the lint logic. This function encapsualtes
+#'   the logic to pull out the matched call in common situations.
+#'
+#' @param expr An
```
Comment (AshesITR) [inline R/xp_utils.R:53]: Cut off

File: NEWS.md (hunk 1 above)
Comment (MichaelChirico) [reply NEWS.md]: bad merge... this item is at L13 now

**Review by AshesITR — APPROVED** [review]

**Pull request merged**